android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -115,27 +125,29 @@ internal data class SuggestionDetails(
     val fullKeyword: String,
     val isSponsored: Boolean,
     val icon: List<UByte>?,
-    val clickInfo: FxSuggestClickInfo? = null,
+    val clickInfo: FxSuggestInteractionInfo? = null,
+    val impressionInfo: FxSuggestInteractionInfo? = null,
 )
 
 /**
- * Collective of fields required for fxsuggest click telemetry
+ * Additional information about a Firefox Suggest [AwesomeBar.Suggestion] to record in telemetry when the user
+ * interacts with the suggestion.
  */
-sealed interface FxSuggestClickInfo {
+sealed interface FxSuggestInteractionInfo {
     /**
-     * AMP related fields for fx suggest click telemetry
+     * Interaction information for a sponsored Firefox Suggest search suggestion from AMP.
      *
      * @param blockId A unique identifier for the suggestion.
      * @param advertiser The name of the advertiser providing the sponsored suggestion.
-     * @param clickUrl The url to report the click to.
+     * @param reportingUrl The url to report the click or impression to.
      * @param iabCategory The categorization of the suggestion.
      * @param contextId The contextual services user identifier.
      */
     data class Amp(
         val blockId: Long,
         val advertiser: String,
-        val clickUrl: String,
+        val reportingUrl: String,
```

Comment:
@tiftran Sorry I suggested changing this from `reportingUrl` to `clickUrl` in your patch, only to change it back! 😬 I was thinking of having a separate `sealed interface FxSuggestImpressionInfo` to store info for impressions, but realized the fields are virtually identical.